### DISCUSSION TITLE
"cannot move across collection"

### DISCUSSION THREAD

**Discussion by Material-Scientist** [seaweedfs/seaweedfs#5920] [Q&A] [answered]
I have a weed mount at `/root/SWFS`, and I'm running a jupyterlab server at `/root`.
Inside jupyterlab, I navigate to `SWFS/buckets/default/codebase/mi/nbs`, and create a file `untitled.txt`, and delete it again. 

The file creation causes no issue, but the deletion fails when performed from inside jupyterlab:
```
Aug 19 21:13:43 SMTRL-01 seaweedfs-mount[996]: I0819 21:13:43.123008 wfs_filer_client.go:31 WithFilerClient 0 10.0.0.4:18888: dir Rename /buckets/default/codebase/mi/nbs/untitled.txt => /.Trash-0/files/untitled.txt receive: rpc error: code = Unknown desc = can not move across collection default =>
Aug 19 21:13:43 SMTRL-01 seaweedfs-mount[996]: I0819 21:13:43.123096 weedfs_rename.go:210 Link: dir Rename /buckets/default/codebase/mi/nbs/untitled.txt => /.Trash-0/files/untitled.txt receive: rpc error: code = Unknown desc = can not move across collection default =>
```

However, when deleting the file via CLI by browsing to `/root/SWFS/buckets/default/codebase/mi/nbs` and deleting with `rm -rf untitled.txt`, no issue arises, and the file is deleted.

What's happening here? Why can it be deleted via CLI, but not from within jupyterlab?

Additional context:
Command for jupyterlab: `jupyter lab --no-browser --port=8889 --allow-root`

**Comment by chrislusf** [comment]
Jupyter seems just does a rename to move the file to a trash folder.

The files in the bucket are in its own collection. So rename didn't work if moving outside the collection.

**Reply by Material-Scientist** [reply, marked as answer]
Yes, seems like jupyter sends files to OS trash on delete: https://github.com/jupyter/notebook/pull/1968

Hm, okay I tried assigning the `.Trash-0` directory to the default collection, using `fs.configure -locationPrefix=/.Trash-0/ -collection=default -apply`, but I'm still getting the same error.

They should be in the same collection now; why does it still fail?

A workaround seems to be `c.FileContentsManager.delete_to_trash=False` in the jupyter config.

However, what if other applications from various collections send to trash? Will SeaweedFS not be able to handle those either?

BTW, this is running on 3.71.

**Reply by chrislusf** [reply]
Specifically, the rename is hard-coded for buckets, where each bucket uses one collection. 

https://github.com/seaweedfs/seaweedfs/blob/master/weed/filer/filer_rename.go#L19

This would make bucket removal instant, without checking any other files referencing the bucket collection.
